bigquery/google/cloud/bigquery/query.py

```diff
@@ -31,7 +31,6 @@ class _SyncQueryConfiguration(object):
     Values which are ``None`` -> server defaults.
     """
     _default_dataset = None
-    _dry_run = None
```

Comment:
Your mistake is in assuming there is any logical consistency between the synchronous query request and the asynchronous query job. :)